drivers/stepper/gpio_stepper_controller.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_steppers
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/drivers/stepper.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+
+LOG_MODULE_REGISTER(gpio_stepper_motor_controller, CONFIG_STEPPER_LOG_LEVEL);
+
+#define MAX_MICRO_STEP_RES STEPPER_MICRO_STEP_2
+#define NUM_CONTROL_PINS   4
+
+static const uint8_t gpio_stepper_half_step_lookup_table[NUM_CONTROL_PINS *
```

Comment:
These are static so you can drop the namespace to keep tve names shorter, for example `half_step_lookup_table[]`